docs/streams/developer-guide/config-streams.html

```diff
@@ -1010,6 +1016,18 @@ <h4><a class="toc-backref" href="#id31">topology.optimization</a><a class="heade
           </p>
         </div></blockquote>
     </div>
+    <div class="section" id="windowed.inner.class.serde">
+      <h4><a class="toc-backref" href="#id31">windowed.inner.class.serde</a><a class="headerlink" href="#windowed.inner.class.serde" title="Permalink to this headline"></a></h4>
+      <blockquote>
+        <div>
+          <p>
+            Serde for the inner class of a windowed record. Must implement the org.apache.kafka.common.serialization.Serde interface.
+          </p>
+          <p>
+            Note that setting this config in KafkaStreams application would result in an error as it is meant to be used only from Plain consumer client.
```

Comment:
This sentence confuses me. I realize that it's also in the Java code, but not sure what it actually means? Maybe @ableegoldman who reviewed the original PR can help out?

Reply:
Bear with me here because it's kind of a long story in full and I only half remember it...and also this statement is only half true. A more accurate statement would be something like this:

"Note that this config is only used by plain consumer/producer clients that set a windowed de/serializer type via configs. For Kafka Streams applications that deal with windowed types you must pass in the inner serde type when you instantiate the windowed serde object for your topology"

In other words, there's no concept of a "default windowed serde" type in Streams because you always have to pass in the actual Serde object, and need to pass in the window size and inner serde type at that time. The same can also be done for plain producer/consumer clients but someone pointed out that the console clients have to be configured via properties, which means they can only invoke the default constructor based on the class name passed in for the key/value de/serializer configs. Which meant we needed to add the window size and inner serde type as configs that could be used to configure a windowed Serde that was instantiated via default constructor

I guess we should update the docs string in StreamsConfig as well. I assume this error slipped in because of the window.size config, which _is_ only needed for the consumer client (since the producer client throws away the window size when encoding a windowed type). But both producer and consumer need to know the inner serde type

Reply:
Thank you Sophie! Changed the sentence.

Reply:
Going back to the [blog post](https://www.confluent.io/blog/apache-kafka-3-4-0-new-features-and-updates/) I don't know where the cache.size metric is coming from-- because in the quoted [KIP-770](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=186878390) it appears to be `cache-size-bytes-total `.

Reply:
>  Are you proposing to move it to ClientConfig or just not .define it in any class? In the end, we added it via KIP and would need to move/remove it via KIP, and I'm not sure it's worth doing a KIP over. We can just leave it out of the Streams config docs

Yes, the idea was to maybe do a small KIP and remove it from `StreamsConfig` as it does not belong there. If Lucia is interested in doing it -- I agree it's not super critical, and just removing from the docs is also fine with me.

About `window.inner.serde.class` it is indeed used by Kafka Streams itself (cf `TimeWindowedSerializer` and others). If one would set default serde to `TimeWindowedSerializer` it would be used, so I think it's ok to have it in the docs, and also correct to have it as `StreamsConfig` registered via `define()`.

Reply:
I just did some more digging, and now I actually think that @ableegoldman is right, we might want to treat `windowed.inner.serde.class` similar to `window.size`... (ie, maybe remove from StreamsConfig -- we could add this to the KIP Lucia started).

I also understand now, why the docs says, using it would result in an error (for both configs): Kafka Streams will always pass window-size and inner-serde via the _constructor_ and we will also verify that we don't get an parameter set twice (or zero time), and throw an error inside `configure()` method of the windowed serdes...

Thus, we might want to not add `windowed.inner.serde.class` to the docs in this PR to begin with...

Sorry for the back and forth. Reading and understanding code is hard...